### ISSUE TITLE
GTK/GLib event dispatching becomes unresponsive

### ISSUE THREAD

**Issue by gogins** [gogins/webkit-opcodes#8]
With a high density performance, the HTML UI becomes unresponsive after 15 seconds or so. I suspect a data race/deadlock.

Running with callgrind shows no problem, the UI continues to respond. I will try a short sleep in between polls to the event dispatcher.

**Comment by gogins** [comment]
It seems this is probably caused by the `webkit_run_javascript opcode`. If that's not used much, then the UI remains completely responsive through a dense performance.

This gives me hope that these opcodes will after all be useful. I might be able to completely fix the problem, and if not, then simply not logging Csound diagnostics on the Web page should adequately work around the problem.

**Comment by gogins** [comment]
I have corrected the code for these opcodes and all exceptions and warnings are now gone. However, the problem of the unresponsive UI remains. I have concluded that the problem occurs when the GTK `webkit_web_view_run_javascript` function is called too frequently. If this function is not used for sending Csound diagnostic messages to the Web page, the UI remains responsive through dense performances.

I am closing this issue. I will amend the examples and documentation to reflect what I have learned.
